react/features/prejoin/components/country-picker/CountryPicker.js

```diff
@@ -0,0 +1,245 @@
+// @flow
+
+import React, { Component } from 'react';
+import InlineDialog from '@atlaskit/inline-dialog';
+import { setDialOutCountry, setDialOutNumber } from '../../actions';
+import CountrySelector from './CountrySelector';
+import CountryDropDown from './CountryDropdown';
+import { getDialOutCountry, getDialOutNumber } from '../../functions';
+import { connect } from '../../../base/redux';
+import { getCountryFromDialCodeText } from '../../utils';
+
+const PREFIX_REG = /^(00)|\+/;
+
+type Props = {
+
+    /**
+     * The country to dial out to.
+     */
+    dialOutCountry: { name: string, dialCode: string, code: string },
+
+    /**
+     * The number to dial out to.
+     */
+    dialOutNumber: string,
+
+    /**
+     * Handler used when user presses 'Enter'.
+     */
+    onSubmit: Function,
+
+    /**
+     * Sets the dial out number.
+     */
+    setDialOutNumber: Function,
+
+    /**
+     * Sets the dial out country.
+     */
+    setDialOutCountry: Function,
+};
+
+type State = {
+
+    /**
+     * If the country picker is open or not.
+     */
+    isOpen: boolean,
+
+    /**
+     * The value of the input.
+     */
+    value: string
+}
+
+/**
+ * This component displays a country picker with an input for the phone number.
+ */
+class CountryPicker extends Component<Props, State> {
```

Comment:
Is there a reason not to use `PureComponent`? It's not a must I know but recently we started to use them to optimize rendering. (same for other components)

Reply:
Done.